### DISCUSSION TITLE
Using personal turbine models - !input

### DISCUSSION THREAD

**Discussion by RT96-stu** [NatLabRockies/floris#816] [Q&A] [answered]
Hi ! 

I would like to use FLORIS on a lot of wind farms with plenty turbine types.
Therefore, I tried to automate the creation of inputs.

I read in an other discussion that you can use personal turbine models as long as you write 
- !input /path/to/turbine/model.yaml 
in the input file. But I didn't manage to write !input without quote marks. I tried the FLORIS fi.reinitialize and yaml.safe_dump. Is there a way to make python write it in the input file ?

I'd prefer a solution without changing the turbine_library, since I work in a share Virtual Environment and it takes time to update libraries.

Thank you in advance

**Comment by misi9170** [comment, marked as answer]
Hi @RT96-stu, to use the !include (not !input, just to be clear) it's easiest to modify the model yaml directly. For instance, try modifying the relevant part of the floris input yaml to something like:

```
farm:
  layout_x:
  - 0.0
  - 630.0
  - 1260.0
  .
  .
  .
  - 5000.0
  - 10000.0
  .
  .
  .
  - 10630.0
  layout_y:
  - 0.0
  - 0.0
  - 0.0
  .
  .
  .
  - 0.0
  - 1000.0
  .
  .
  .
  - 1000.0
  turbine_type:
  - !include path/to/turbine_type_1
  - !include path/to/turbine_type_1
  - !include path/to/turbine_type_1
  . 
  .
  . 
  - !include path/to/turbine_type_1
  - !include path/to/turbine_type_2
  .
  .
  . 
  - !include path/to/turbine_type_2
  ```

This is explained more in #766 (which I think you were alluding to). To use the `!include` notation, you need to modify the input yaml itself---this notation can't be passed to `fi.reinitialize()`. If you want to pass the turbine types to `fi.reinitialize()` directly, you'll need to first load the turbine models to a dictionaries (e.g. `turbine_dict_1`, `turbine_dict_2`) and then pass those dictionaries to the reinitialize method (e.g. `fi.reinitialize(layout_x=[0, 1000], layout_y=[0, 0], turbine_type=[turbine_dict_1, turbine_dict_2])`)

**Reply by RT96-stu** [reply]
Hi @misi9170 !
Thanks for the answer. 
I managed to use the - !include method you described. I tried to write it with the yaml library on python, which forced me to put the !include within quote marks. With the built-in write method, it works !
